dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/DotContentletTransformerImpl.java

```diff
@@ -0,0 +1,115 @@
+package com.dotmarketing.portlets.contentlet.transform;
+
+import static com.dotmarketing.portlets.contentlet.transform.strategy.TransformOptions.BINARIES;
+import static com.dotmarketing.portlets.contentlet.transform.strategy.TransformOptions.COMMON_PROPS;
+import static com.dotmarketing.portlets.contentlet.transform.strategy.TransformOptions.CONSTANTS;
+import static com.dotmarketing.portlets.contentlet.transform.strategy.TransformOptions.VERSION_INFO;
+import static com.dotmarketing.util.UtilMethods.isSet;
+
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.util.CollectionsUtils;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.transform.strategy.StrategyResolver;
+import com.dotmarketing.portlets.contentlet.transform.strategy.TransformOptions;
+import com.dotmarketing.portlets.contentlet.transform.strategy.TransformToolbox;
+import com.dotmarketing.util.Logger;
+import com.google.common.annotations.VisibleForTesting;
+import com.liferay.portal.model.User;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class intents to be the single point of transformation logic
+ * Transformation basically takes place in strategies that are plugged based on the options submitted
+ * or the ContentType associated to the the Content.
+ */
+class DotContentletTransformerImpl implements DotContentletTransformer {
+
+    static final Set<TransformOptions> defaultOptions = EnumSet.of(
+            COMMON_PROPS, CONSTANTS, VERSION_INFO, BINARIES
+    );
+
+    private User user;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Private field 'user' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)